File: app/src/main/kotlin/com/sealkrach/politirush/ProgressStore.kt
package com.sealkrach.politirush

import android.content.Context
import com.sealkrach.politirush.core.MetaProgress
import com.sealkrach.politirush.core.Upgrade

/**
 * Persistance minimale de la progression dans les SharedPreferences.
 * À remplacer par DataStore (et une sauvegarde cloud) quand le jeu grossira.
 */
class ProgressStore(context: Context) {
    private val prefs = context.getSharedPreferences("politirush_progress", Context.MODE_PRIVATE)

    fun load(): MetaProgress = MetaProgress(
        coins = prefs.getInt("coins", 0),
        bestScore = prefs.getInt("bestScore", 0),
        bestWave = prefs.getInt("bestWave", 0),
        runs = prefs.getInt("runs", 0),
        levels = Upgrade.entries.associateWith { prefs.getInt("lvl_${it.name}", 0) }.filterValues { it > 0 },
        unlockedHeroes = prefs.getStringSet("heroes", setOf("rambeau")) ?: setOf("rambeau"),
    )

    fun save(p: MetaProgress) {
        prefs.edit().apply {
            putInt("coins", p.coins)
            putInt("bestScore", p.bestScore)
            putInt("bestWave", p.bestWave)
            putInt("runs", p.runs)
            Upgrade.entries.forEach { putInt("lvl_${it.name}", p.level(it)) }
            putStringSet("heroes", p.unlockedHeroes)
        }.apply()
    }
}

File: Assets/PolitiRush/Runtime/ProgressStore.cs
using PolitiRush.Core;
using UnityEngine;

namespace PolitiRush.Runtime
{
    /// <summary>Progression persistante dans PlayerPrefs (équivalent de ProgressStore.kt).</summary>
    public static class ProgressStore
    {
        public static MetaProgress Load()
        {
            var m = new MetaProgress { Coins = PlayerPrefs.GetInt("coins", 0), BestScore = PlayerPrefs.GetInt("bestScore", 0), BestWave = PlayerPrefs.GetInt("bestWave", 0), Runs = PlayerPrefs.GetInt("runs", 0) };
            foreach (var u in UpgradeInfo.All) { int l = PlayerPrefs.GetInt("lvl_" + u, 0); if (l > 0) m.Levels[u] = l; }
            return m;
        }

        public static void Save(MetaProgress m)
        {
            PlayerPrefs.SetInt("coins", m.Coins); PlayerPrefs.SetInt("bestScore", m.BestScore); PlayerPrefs.SetInt("bestWave", m.BestWave); PlayerPrefs.SetInt("runs", m.Runs);
            foreach (var u in UpgradeInfo.All) PlayerPrefs.SetInt("lvl_" + u, m.Level(u));
            PlayerPrefs.Save();
        }
    }
}
